File: advent_of_code/2024/02/first.cs
using System;
using System.IO;
using System.Collections.Generic;
using System.Linq;


class Program
{
    static void Main()
    {
        string filePath = "input.txt";

        int ans = 0;

        try
        {
            string[] lines = File.ReadAllLines(filePath);

            foreach (string line in lines)
            {
                string[] parts = line.Split(" ");
                List<int> numbers = parts.Select(int.Parse).ToList();
                if (numbers[0] == numbers[1]) continue;
                bool increase = numbers[0] < numbers[1];
                bool correct = true;
                for (int i = 1; i < numbers.Count; i++)
                {
                    if (increase && (numbers[i] - numbers[i - 1] > 3 || numbers[i] - numbers[i - 1] < 1))
                    {
                        correct = false;
                        break;
                    }
                    else if (!increase && (numbers[i - 1] - numbers[i] > 3 || numbers[i - 1] - numbers[i] < 1))
                    {
                        correct = false;
                        break;
                    }
                }
                if (correct) ans++;
            }
        }
        catch (Exception e)
        {
            Console.WriteLine(e.Message);
        }

        Console.WriteLine(ans);
    }
}

File: advent_of_code/2025/03/first.js
const fs = require('node:fs')

const input = fs.readFileSync('./input.txt', 'utf8')

const lines = input.split('\n')

let ans = 0

let curr = 50

for (const line of lines) {
  let d1 = -1
  let d2 = -1
  let p1 = -1
  for (let i = 0; i < line.length - 1; i++) {
    const d = Number(line[i])
    if (d > d1) {
      d1 = d
      p1 = i
    }
  }
  for (let i = p1 + 1; i < line.length; i++) {
    const d = Number(line[i])
    if (d > d2) {
      d2 = d
      p2 = i
    }
  }

  ans += d1 * 10 + d2
}


console.log(ans)